src/util/include/util/GListView.h

```diff
@@ -16,10 +16,15 @@
 
 #include <glib.h>
 
-template <typename T>
-
-struct GListView {
+template <class ListType, typename T>
+struct ListView {
     struct GListViewIter {
+        using iterator_category = std::forward_iterator_tag;
+        using value_type = T;
+        using pointer = T*;
+        using reference = T&;
+        using difference_type = std::ptrdiff_t;
+
```

Comment:
It is to meet the requirement for iterators (cf e.g. https://en.cppreference.com/w/cpp/iterator/iterator_traits).
With those, we can use the functions of the std::algorithm library. Otherwise, you'll get a compilation error.

Reply:
I don't get a compilation error when removing those lines with g++ 12.3 (even after wiping the build folder completely), but thanks for the explanation.